libs/prelude/Prelude/Functor.idr

```diff
@@ -22,3 +23,5 @@ infixl 4 <$>
 (<$>) : Functor f => (func : a -> b) -> f a -> f b
 func <$> x = map func x
 
+Functor (Pair a) where
+  map f (x, y) = (x, f y)
```

Comment:
On that note, shouldn't there be a homogenus tuple type, or are we assuming that `Vect 2 a` is good enough for that purpose?